network-api/networkapi/wagtailpages/pagemodels/blog/blog.py

```diff
@@ -223,7 +223,11 @@ def get_context(self, request):
             # see that same padded list during preview, but *without* actually updating
             # the model, so we control this property at render context retrieval time:
             related_posts = related_posts + self.get_missing_related_posts()
-        context['related_posts'] = related_posts
+
+        # Make sure to filter out any None entries, which may happen if the
+        # self.ensure_related_posts() function was unable to find any related
+        # posts to pad the related posts set with.
+        context['related_posts'] = list(filter(None, related_posts))
```

Comment:
This is mostly matter of taste I guess, but I was wondering if we should use list comprehension rather than the list -> filter -> list nesting. I find them a bit more readble 😅 

(I can't find proof right now, but I think they are also supposed to be faster)

Reply:
The reason I'd use this rather than `[x for x in related_posts if x is not None]` is the increased potential-for-bugs surface: the list comprehention has three places to typo the anonymous variable, as opposed to "there isn't even a variable to typo" in the filter call.

Reply:
Fair 👍